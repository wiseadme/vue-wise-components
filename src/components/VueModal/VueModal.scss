.vue-modal {
  @include flex(center, center, column);
  position: absolute;
  width: 250px;
  min-width: 250px;
  background: $white;
  border-radius: 6px;
  box-shadow: $shadow;
  z-index: 3;

  &-container {
    @include flex(center, center);
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    z-index: 101;
  }

  &__window {
    @include flex(center, center, column);
    position: absolute;
    width: 250px;
    min-width: 250px;
    background: $white;
    border-radius: 6px;
    z-index: 3;
    box-shadow: $shadow;
  }

  &__body {
    @include flex(center, center, column);
    text-align: center;
    width: 100%;
    height: 200px;
    min-height: 200px;
    font-weight: 500;
    font-size: 17px;
    line-height: 20px;
  }

  &__footer {
    @include flex(space-evenly, center);
    width: 100%;
  }
}
