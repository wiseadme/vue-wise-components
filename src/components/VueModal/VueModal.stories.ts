// import VueModal from '@/components/VueModal'
import { storiesOf } from '@storybook/vue'

storiesOf('VueModal', module)
  .add('overlayable', () => ({
    data() {
      return {
        showModal: true
      }
    },
    template: `
      <div class="wrap">
        <vue-button text="click" class="green--accent-3"  @click="showModal = true"/>
        <vue-modal
          :overlay="true"
          transition="scaleIn"
          v-model="showModal"
          v-if="showModal"
        >
          <div
            slot="header"
            class="modal-header cyan--darken-2"
          >
            warning
          </div>
          <template slot="body">
            <span>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab ad amet, aspernatur, consectetur deserunt eius ex inventore iure, iusto minima officia quisquam reprehenderit repudiandae temporibus tenetur. Ab alias aliquam aspernatur beatae commodi delectus fuga maiores, nesciunt, provident quaerat, rem sit.</span>
          </template>
          <div class="btn-wrap" slot="footer">
            <vue-button
              text="ok"
              type="primary"
              @click="showModal = !showModal"
            />
          </div>
        </vue-modal>
      </div>`
  }))
  .add('without overlay', () => ({
    data() {
      return {
        showModal: true
      }
    },
    template: `
      <div class="wrap">
        <vue-button text="click" class="teal--accent-3"  @click="showModal = true"/>
        <vue-modal
          :overlay="false"
          transition="scaleIn"
          v-model="showModal"
          v-if="showModal"
        >
          <div
            slot="header"
            class="modal-header cyan--darken-2"
          >
            warning
          </div>
          <template slot="body">
            <span>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab ad amet, aspernatur, consectetur deserunt eius ex inventore iure, iusto minima officia quisquam reprehenderit repudiandae temporibus tenetur. Ab alias aliquam aspernatur beatae commodi delectus fuga maiores, nesciunt, provident quaerat, rem sit.</span>
          </template>
          <div class="btn-wrap" slot="footer">
            <vue-button
              text="ok"
              type="primary"
              @click="showModal = !showModal"
            />
          </div>
        </vue-modal>
      </div>`
  }))
