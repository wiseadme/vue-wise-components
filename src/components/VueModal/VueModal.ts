// Styles
import './VueModal.scss'

// Types
import { VNode } from 'vue/types'
import { mixins } from '@/utils/mixins'

// Mixins
import Overlayable from '@/mixins/Overlayable'

export default mixins(Overlayable).extend({
  name: 'VueModal',

  props: {
    transition: String,
    overlay: Boolean,
    value: Boolean,
    width: [String, Number],
  },

  beforeMount() {
    this.overlay && this.$nextTick(() => {
      this.genOverlay()
    })
  },

  beforeDestroy() {
    this.overlay && this.removeOverlay()
  },

  watch: {
    value() {
      this.overlay && this.toggleOverlay()
    },
  },

  methods: {
    toggleOverlay() {
      this.hideOverlay = !this.hideOverlay
    },

    genTransition(content: VNode | VNode[]): VNode | null {
      if (!this.transition) return content as VNode

      return this.$createElement('transition', {
        props: {
          name: this.transition,
          appear: true,
        }
      }, [content])
    },
  },

  computed: {
    container(): VNode {
      return this.$createElement('div', {
        staticClass: 'vue-modal-container',
        directives: [
          {
            name: 'show',
            value: this.value
          }
        ],
        ref: 'modal'
      }, [])
    },

    modal(): VNode {
      return this.$createElement('div', {
        staticClass: 'vue-modal',
        style: { maxWidth: this.width + 'px' },
      }, this.$slots.default)
    }
  },

  render(h): VNode {
    this.container.children!.push(this.modal)

    return h('transition', {}, [
      this.genTransition(this.container)
    ])
  }
})
