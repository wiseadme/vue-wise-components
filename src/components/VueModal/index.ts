import VueModal from './VueModal'

export { VueModal }
export default VueModal
