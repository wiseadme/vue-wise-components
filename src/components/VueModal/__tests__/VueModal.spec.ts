import { mount } from '@vue/test-utils'
import VueModal from '@/components/VueModal'
// import VueOverlay from '@/components/VueOverlay'

describe('VueModal.ts', () => {
  let vm: any | null = null
  beforeEach(() => {
    vm = mount(VueModal,{
      propsData: {
        overlay: true,
        valid: true
      }
    })
  })
  it('show overlay if passed to true', () => {
    expect(vm.element).toMatchSnapshot()
  })
})
