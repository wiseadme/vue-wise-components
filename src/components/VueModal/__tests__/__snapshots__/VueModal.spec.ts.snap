// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueModal.ts show overlay if passed to true 1`] = `
<transition-stub
  valid="true"
>
  <div
    class="vue-modal-container"
    style="display: none;"
  >
    <div
      class="vue-modal"
    />
  </div>
</transition-stub>
`;
